mrmustard/math/caching.py

```diff
@@ -33,7 +36,7 @@ def cached_wrapper(hashable_array, cutoff):
 
     @wraps(fn)
     def wrapper(tensor, cutoff):
-        return cached_wrapper(tuple(tensor.numpy()), cutoff)
+        return cached_wrapper(tuple(math.asnumpy(tensor)), cutoff)
```

Comment:
good catch

Reply:
Eh, with two backends all of these lines error